src/lib/pages/frontsite/Documents/DocumentDetails/DocumentItems/DocumentItem.js

```diff
@@ -2,59 +2,169 @@ import { getDisplayVal, invenioConfig } from '@config';
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
 import Overridable from 'react-overridable';
-import { Grid, List } from 'semantic-ui-react';
+import { Button, Card, Table } from 'semantic-ui-react';
+import { Media } from '@components/Media';
 
 class DocumentItem extends Component {
-  render() {
-    const { item } = this.props;
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isShowingAll: false,
+      itemAmountLimit: 5,
+    };
+  }
+
+  get moreItemsToLoad() {
+    const { physicalCopies } = this.props;
+    const { itemAmountLimit } = this.state;
+
+    return physicalCopies.length > itemAmountLimit;
+  }
+
+  sliceCopiesArray = (array) => array.slice(0, this.state.itemAmountLimit);
+
+  toggleAllCopies = () =>
+    this.setState({ isShowingAll: !this.state.isShowingAll });
+
+  itemStatus = (item) => ({
+    itemIsOnShelf: () =>
+      invenioConfig.ITEMS.canCirculateStatuses.includes(item.status) &&
+      !item.circulation,
+
+    itemIsForReference: () =>
+      invenioConfig.ITEMS.referenceStatuses.includes(item.status),
+
+    itemIsOnLoan: () =>
+      invenioConfig.ITEMS.canCirculateStatuses.includes(item.status) &&
+      item.circulation,
+  });
+
+  statusLabel = (item) => {
+    const onShelf = this.itemStatus(item).itemIsOnShelf();
```

Comment:
Nice solution, I find quite readable! :) I would remove the extra `item` prefix so that you can read it:
`this.itemStatus(item).isOnShelf()`